src/ayab/knitter.cpp

```diff
@@ -118,34 +116,49 @@ void Knitter::isr() {
   m_carriage = GlobalEncoders::getCarriage();
 }
 
+Err_t Knitter::initMachine(Machine_t machineType) {
+  if (GlobalFsm::getState() != s_wait_for_machine) {
+    return WRONG_MACHINE_STATE;
+  }
+  if (machineType == NoMachine) {
+    return NO_MACHINE_TYPE;
+  }
+  if (machineType >= NUM_MACHINES) {
+    return MACHINE_TYPE_INVALID;
+  }
+
+  m_machineType = machineType;
+
+  // Now that we have enough start state, we can set up interrupts
+  setUpInterrupt();
```

Comment:
Should this (enable interrupts) not move after encoder/FSM init L135/136 ?

Reply:
The interrupt used to be setup long before the encoder, but yea, for consistency sake, it probably should be.